Xamarin.Forms.Core/Layout.cs

```diff
@@ -318,69 +318,88 @@ internal static void LayoutChildIntoBoundingRegion(View child, Rectangle region,
+						(trigger == InvalidationTrigger.SizeRequestChanged && view.ComputedConstraint == LayoutConstraint.Fixed))
+					{
+						return;
+					}
+					if (trigger == InvalidationTrigger.HorizontalOptionsChanged || trigger == InvalidationTrigger.VerticalOptionsChanged)
+					{
+						ComputeConstraintForView(view);
+					}
 				}
-			}
 
-			_allocatedFlag = false;
-			if (trigger == InvalidationTrigger.RendererReady)
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
-			}
-			else
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
-			}
-
-			s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
-			if (!s_relayoutInProgress)
-			{
-				s_relayoutInProgress = true;
-
-				// Rather than recomputing the layout for each change as it happens, we accumulate them in 
-				// s_resolutionList and schedule a single layout update operation to handle them all at once.
-				// This avoids a lot of unnecessary layout operations if something is triggering many property
-				// changes at once (e.g., a BindingContext change)
+				_allocatedFlag = false;
+				if (trigger == InvalidationTrigger.RendererReady)
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
+				}
+				else
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
+				}
 
-				Device.BeginInvokeOnMainThread(() =>
+				s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
+				if (!s_relayoutInProgress)
 				{
-					// if thread safety mattered we would need to lock this and compareexchange above
-					IList<KeyValuePair<Layout, int>> copy = s_resolutionList;
-					s_resolutionList = new List<KeyValuePair<Layout, int>>();
-					s_relayoutInProgress = false;
+					s_relayoutInProgress = true;
 
-					foreach (KeyValuePair<Layout, int> kvp in copy.OrderBy(kvp => kvp.Value))
+					// Rather than recomputing the layout for each change as it happens, we accumulate them in 
+					// s_resolutionList and schedule a single layout update operation to handle them all at once.
+					// This avoids a lot of unnecessary layout operations if something is triggering many property
+					// changes at once (e.g., a BindingContext change)
+
+					if (Dispatcher != null)
 					{
-						Layout layout = kvp.Key;
-						double width = layout.Width, height = layout.Height;
-						if (!layout._allocatedFlag && width >= 0 && height >= 0)
+						Dispatcher.BeginInvokeOnMainThread(() =>
 						{
-							layout.SizeAllocated(width, height);
-						}
+							UpdateContext();
+						});
 					}
-				});
+					else
+					{
+						Device.BeginInvokeOnMainThread(() => 
+						{
+							UpdateContext();
+						});
+					}			
+				}
+			}
+		}
+
+		internal void UpdateContext()
+		{
+			// if thread safety mattered we would need to lock this and compareexchange above
+			IList<KeyValuePair<Layout, int>> copy = s_resolutionList;
```

Comment:
Can here immediately sort it. Before sent it in `foreach` cycle?
```suggestion
			var copy = s_resolutionList.OrderBy(kvp => kvp.Value);
```